pkg/nfd-worker/nfd-worker.go

```diff
@@ -271,6 +314,43 @@ func (w *nfdWorker) Run() error {
 	labelTrigger.Reset(w.config.Core.SleepInterval.Duration)
 	defer labelTrigger.Stop()
 
+	// Create owner ref
+	ownerReference := []metav1.OwnerReference{}
+	// Get pod owner reference
+	podName := os.Getenv("POD_NAME")
+
+	selfPod, err := w.k8sClient.CoreV1().Pods(w.kubernetesNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
```

Comment:
we should log (and fail imo) if other error happened.

Reply:
I think it's fine to bail out in case of an error (any error) occurred.

Reply:
how about now?